app/services/cancellation_fee_calculator.rb

```diff
@@ -12,10 +12,10 @@ def fee
 
   def calculated_fee
     unless defined?(@calculated_fee)
-      original_canceled_at = @reservation.canceled_at
-      @reservation.canceled_at = Time.current
+      original_canceled_at = @reservation.order_detail.canceled_at
```

Comment:
I would lean towards cleaning this up in a later refactoring, I think. I hesitate to add much more to this PR. Multiple signups might be a good opportunity.